### ISSUE TITLE
[ntuple] Check the usage of `Long_t` in RField (size of the type might differ between 32- and 64-bit platforms)

### ISSUE THREAD

**Issue by jalopezg-r00t** [root-project/root#11616]
- [X] Checked for duplicates

### Describe the bug
As discussed in PR #11525 ([here](https://github.com/root-project/root/pull/11525#discussion_r997329752)), the type `Long_t` might have different sizes on 32- or 64- bit platforms.

Both, `RFieldBase::Create()` (see [here](https://github.com/root-project/root/blob/master/tree/ntuple/v7/src/RField.cxx#L77)), and the aforementioned PR incorrectly assume this to be equivalent to `std::int64_t` - which, again, might or not be the case.

**Comment by jblomer** [comment]
Fixed by #16039

**Comment by hahnjo** [comment]
Yes, https://github.com/root-project/root/pull/16039 takes care of `Long_t`, but we still need to update `RProxiedCollectionField` to remove the hard-coded fixed width integer types.

**Comment by hahnjo** [comment]
Yes, I believe with

> we still need to update `RProxiedCollectionField` to remove the hard-coded fixed width integer types.

I mean this part of the constructor:
https://github.com/root-project/root/blob/05e4403401f8be27cf58fce0fc0410e781d5a1ff/tree/ntuple/v7/src/RFieldMeta.cxx#L643-L658

The solution may be as easy as constructing `RField<long>` and so on, and let `RIntegralTypeMap` figure out the rest. That needs a closer look though...
